src/sinks/azure_common/config.rs

```diff
@@ -121,34 +121,44 @@ pub fn build_client(
     connection_string: Option<String>,
     storage_account: Option<String>,
     container_name: String,
+    blob_endpoint: Option<String>,
 ) -> crate::Result<Arc<ContainerClient>> {
     let client;
     match (connection_string, storage_account) {
         (Some(connection_string_p), None) => {
             let connection_string = ConnectionString::new(&connection_string_p)?;
-            let mut builder = ClientBuilder::new(
-                connection_string
-                    .account_name
-                    .ok_or("Account name missing in connection string")?,
-                connection_string.storage_credentials()?,
-            )
-            .retry(RetryOptions::none());
-            if let Some(uri) = connection_string.blob_endpoint {
```

Comment:
I refactored this code to remove a need for a mutable builder, and to avoid creation of a CloudLocation and then throwing it away by overwriting it with `build.cloud_location`.